client/src/main/java/io/pravega/client/control/impl/Controller.java

```diff
@@ -364,6 +365,16 @@
      */
     CompletableFuture<Transaction.Status> checkTransactionStatus(final Stream stream, final UUID txId);
 
+    /**
+     * Get list of TransactionInfo for the Stream having status COMMITTED/ABORTED from most recent batch.
+     * TransactionInfo contains unique transactionId, status of transaction and stream.
+     * This api can return maximum 200 records.
```

Comment:
Where this constant comes from? Shouldn't this be configurable?

Reply:
As per our implementation, a batch can have maximum 200 records. If no of records are more than 200 then new batch will be created and entries will be saved in new completedTransactionsBatch-batch table. This API is fetching entries from recent batch only, so it can return max 200 records.

Reply:
Max txn batch size is 100 and not 200. Pls see: https://github.com/pravega/pravega/blob/9404b59a5d2e06a2792fc38f146f4fca7a0042fd/controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/CommitRequestHandler.java#L63

So your API should not return more than 100 records if it is only looking at the last batch.
It may be good to confirm this by adding a test case.